src/document/range.js

```diff
@@ -72,6 +72,211 @@ CKEDITOR.define( [ 'document/positioniterator', 'document/position' ], ( Positio
 		}
 
 		/**
+		 * Checks whether this {document.Range range} intersects with the given {document.Range range}.
+		 *
+		 * @param {document.Range} otherRange Range to check.
+		 * @returns {boolean} True if ranges intersect.
+		 */
+		intersectsWith( otherRange ) {
```

Comment:
After short investigation ( :) ) it appeared, that `intersectsWith` is used only within `Range` class and it was overcomplicated. I removed it and provided simple condition to check in it's place.

Of course I also removed tests for `intersectsWith` but before I did it, I used the mentioned check with the tests for `intersectsWith` and they were all green.